README.md

```diff
@@ -1,115 +1,624 @@
+- An object-oriented API for all metrics
+- Paired bootstrap resampling and approximate randomization tests for significance
+
+The official version is hosted at <https://github.com/mjpost/sacrebleu>.
+
+# Motivation
+
+Comparing BLEU scores is harder than it should be. Every decoder has its own implementation, often borrowed from Moses, but maybe with subtle changes.
+Moses itself has a number of implementations as standalone scripts, with little indication of how they differ (note: they mostly don't, but `multi-bleu.pl` expects tokenized input). Different flags passed to each of these scripts can produce wide swings in the final score. All of these may handle tokenization in different ways. On top of this, downloading and managing test sets is a moderate annoyance.
+
+Sacre bleu! What a mess.
+
+**SacreBLEU** aims to solve these problems by wrapping the original reference implementation ([Papineni et al., 2002](https://www.aclweb.org/anthology/P02-1040.pdf)) together with other useful features.
+The defaults are set the way that BLEU should be computed, and furthermore, the script outputs a short version string that allows others to know exactly what you did.
+As an added bonus, it automatically downloads and manages test sets for you, so that you can simply tell it to score against `wmt14`, without having to hunt down a path on your local file system.
+It is all designed to take BLEU a little more seriously.
+After all, even with all its problems, BLEU is the default and---admit it---well-loved metric of our entire research community.
+Sacre BLEU.
+
+# Features
+
 - It automatically downloads common WMT test sets and processes them to plain text
 - It produces a short version string that facilitates cross-paper comparisons
 - It properly computes scores on detokenized outputs, using WMT ([Conference on Machine Translation](http://statmt.org/wmt17)) standard tokenization
 - It produces the same values as official script (`mteval-v13a.pl`) used by WMT
 - It outputs the BLEU score without the comma, so you don't have to remove it with `sed` (Looking at you, `multi-bleu.perl`)
 
-The official version is hosted at <https://github.com/mjpost/sacrebleu>.
-
-# QUICK START
+# Installation
 
-Install the Python module (Python 3 only)
+Install the official Python module from PyPI (**Python>=3.6 only**):
 
-    pip3 install sacrebleu
+    pip install sacrebleu
 
 In order to install Japanese tokenizer support through `mecab-python3`, you need to run the
 following command instead, to perform a full installation with dependencies:
 
-    pip3 install sacrebleu[ja]
-
-Alternately, you can install from the source:
+    pip install sacrebleu[ja]
 
-    python3 setup.py install
+# Command-line Usage
 
-This installs a shell script, `sacrebleu`.
-(You can also run `python3 -m sacrebleu`, so long as this root directory is in your `$PYTHONPATH`).
+## Basics
 
 Get a list of available test sets:
 
-    sacrebleu --list
+```
```

Comment:
It is great that all the (old and new) features of SacreBLEU are now well documented.
However, I think the README is now a bit too long and newcomers may get impression that SacreBLEU is too difficult to use.
An alternative would be to break the current content into several pages which will be just linked from the main README.
The README should show perhaps just:
```
$ pip install sacrebleu
$ cat output.detok.txt | sacrebleu -t wmt17 -l en-de
BLEU|nrefs:1|case:mixed|eff:no|tok:13a|smooth:exp|version:2.0.0 = 20.8 54.4/26.6/14.9/8.7 (BP = 1.000 ratio = 1.026 hyp_len = 62880 ref_len = 61287)
```
The detailed documentation of all features could be also maintained as GitHub wiki (as done e.g. in [VW](https://github.com/VowpalWabbit/vowpal_wabbit/wiki)) or ReadTheDocs.

That needs some effort, so maybe it can be done in another PR after this PR is merged.

Within this PR, I would just suggest to delete the output of `sacrebleu --list`.
It will soon become outdated. If we want provide the listing in Markdown,
we could generate it automatically (within deployment) to a separate file.

Reply:
exactly, I didn't have the energy to do the auto generated docs bits for this PR but that should arrive sooner or later.

For the `--list`, since we are doing the PRs and merges, I thought we could ask people to just add a line to that portion in the README whenever they add a new dataset. Isn't that reasonable? It's not that we add a new dataset every now and then.

Reply:
I think its better to ask users to run `sacrebleu --list` themselves and maybe remind them to run `pip install --upgrade sacrebleu` first.
That said, you are right it shouldn't be much work to keep the listing updated (although I would still prefer to have it at a separate page).